lib/charms/prometheus_k8s/v0/prometheus.py

```diff
@@ -542,11 +548,11 @@ def _relation_hosts(self, relation):
             A dictionary that maps unit names to unit addresses for
             the specified relation.
         """
-        hosts = {}
-        for unit in relation.units:
-            if host_address := relation.data[unit].get("prometheus_scrape_host"):
-                hosts[unit.name] = host_address
-        return hosts
+        return {
```

Comment:
I do not think this change is warranted. It is simple enough to read `return hosts` and understand that `hosts` are being returned. I do not think `return {... multi-line dictionary comprehension ... }` improves readability and any (if any) miniscule performance gains are useful here.

Reply:
I did not propose this for reasons of performance. I found myself debugging stuff in this codebase because of list modification via `append`, and I feel safer with a more declarative approach. If you vero this change, I'll revert.